Release/1.2.1

flannooz opened the pull request:
dev
## JIRA

* [Main JIRA ticket](https://jira.softserve.academy/secure/RapidBoard.jspa?rapidView=id)


## Code reviewers

- [ ] @github_username

### Second Level Review

- [ ] @github_username

## Summary of issue

ToDo

## Summary of change

ToDo

## Testing approach

ToDo

## CHECK LIST
- [ ]  СI passed
- [ ]  Сode coverage >=95%
- [ ]  PR is reviewed manually again (to make sure you have 100% ready code)
- [ ]  All reviewers agreed to merge the PR
- [ ]  I've checked new feature as logged in and logged out user if needed
- [ ]  PR meets all conventions


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Introduced a new Docker image and entrypoint script for running database updates.
	- Added support for managing and deploying a separate database update container alongside the main backend.
	- Integrated Discord notifications for deployment events.
- **Improvements**
	- Enhanced deployment pipeline to include security scanning for both backend and database update images.
	- Updated deployment process to handle multiple images and improve deployment control flow.
- **Removals**
	- Removed automatic database migration and data seeding from application startup.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

sashapanasiuk5 committed:
add importing connection string using own appsettings.json file

sashapanasiuk5 committed:
remove applying migrations in backend

sashapanasiuk5 committed:
disable hangfire jobs

Commit:
add dockerfile.dbupdate

Commit:
Merge branch 'feature/issue-1681' of https://github.com/ita-social-projects/StreetCode into feature/issue-1681

Commit:
update dockerfile dbupdate

MariiaNazarchuk committed:
Merge pull request #1812 from ita-social-projects/init_container

add dockerfile.dbupdate

sashapanasiuk5 committed:
fix path of working directory and add reference to scripts folder

Commit:
Add dbupdate image to Jenkinsfile

MariiaNazarchuk committed:
Merge branch 'release/1.2.1' into feature/issue-1681

Commit:
Add TrustServerCertificate option to DbUpdate connection string

sashapanasiuk5 committed:
uncomment hangfire jobs

sashapanasiuk5 committed:
add missing script

sashapanasiuk5 committed:
include entire ScriptsMigration folder

sashapanasiuk5 committed:
fix sql scipt

sashapanasiuk5 committed:
fix appsettings

sashapanasiuk5 committed:
remove old appsetting and add new for integration tests

sashapanasiuk5 committed:
change connection string for tests

YaroslavRosnovskyi committed:
regenerated sql scripts

YaroslavRosnovskyi committed:
changed jenkins configuration

YaroslavRosnovskyi committed:
removed paralel execution

Commit:
Update expression condition

YaroslavRosnovskyi committed:
changed version for dotnet-coverage

YaroslavRosnovskyi committed:
Merge branch 'feature/issue-1681' of https://github.com/ita-social-projects/StreetCode into feature/issue-1681

YaroslavRosnovskyi committed:
revert jenkinsfile

YaroslavRosnovskyi committed:
updated jenkins config

YaroslavRosnovskyi committed:
Revert "updated jenkins config"

This reverts commit fbdfcb1e5a3c29778bee31e09bfd731b4c57b27b.

YaroslavRosnovskyi committed:
changed config

YaroslavRosnovskyi committed:
fixed config

YaroslavRosnovskyi committed:
added no build

Commit:
Add Discord notification

YaroslavRosnovskyi committed:
changed version

Michael-Kolpakov committed:
Merge pull request #2102 from ita-social-projects/feature/issue-1681

Add dbupdate image to Jenkinsfile

YaroslavRosnovskyi committed:
merged 1.2.0 into 1.2.1

YaroslavRosnovskyi committed:
fixed scripts

YaroslavRosnovskyi committed:
added test script

YaroslavRosnovskyi committed:
renamed sql file

YaroslavRosnovskyi committed:
modified dockerfile

YaroslavRosnovskyi committed:
added logs

YaroslavRosnovskyi committed:
added logs

YaroslavRosnovskyi committed:
changed connection string

YaroslavRosnovskyi committed:
modified dockerfile

YaroslavRosnovskyi committed:
modified docker update

YaroslavRosnovskyi committed:
removed local appsettings

YaroslavRosnovskyi committed:
updated Dockerfile dbupdate

Commit:
Add entrypoint.sh for image debugging

Commit:
Add debugging tools and entrypoint.sh

ihor-yatsyshyn committed:
Added test script

ihor-yatsyshyn committed:
Revert "Added test script"

This reverts commit defdde09d6bd17d830a9009f9c8829282ec22cb7.

ihor-yatsyshyn committed:
Add health check commit

ihor-yatsyshyn committed:
Changed test scripts

ihor-yatsyshyn committed:
Deleted additional health check

ihor-yatsyshyn committed:
Reverted changes

ihor-yatsyshyn committed:
Create HealthCheck.sql

ihor-yatsyshyn committed:
Update HealthCheck.sql

ihor-yatsyshyn committed:
Update entrypoint.sh

Commit:
Commit string with mock failure

Commit:
remove entrypoint and tools

Commit:
add copy the published output into the runtime

MariiaNazarchuk committed:
Merge branch 'release/1.2.1' into feature/issue-2024

MariiaNazarchuk committed:
Merge pull request #2443 from ita-social-projects/feature/issue-2024

Add Discord notification

Commit:
make Discord webhook usage safer

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images

Commit:
added trivy scan for docker images backend

flannooz committed:
Merge pull request #2729 from ita-social-projects/feature/issue-1172

Added a Trivy scan for Docker images after the build

flannooz commented on Jenkinsfile:
```diff
@@ -138,46 +141,87 @@ pipeline {
                 script {
                     checkout scmGit(
-                      branches: [[name: 'main']],
+                      branches: [[name: 'feature/add-init-container']],
```
we have to change 'feature/add-init-container' to 'main' there are

LanchevychMaxym replied:
If we merged changes in Streetcode-DevOps repo let's update here to main as well

flannooz replied:
Alright, it's done

LanchevychMaxym commented on Dockerfile.dbupdate:
```diff
@@ -0,0 +1,31 @@
+COPY ./Streetcode/Streetcode.WebApi/appsettings.Staging.json ./DbUpdate/
+COPY ./Streetcode ./Streetcode/
+COPY ./Streetcode/Streetcode.DAL/Persistence/ScriptsMigration /tmp/ScriptsMigration/
+RUN ls -la /tmp/ScriptsMigration
```
probably we can remove this line
```suggestion
```

LanchevychMaxym commented on Jenkinsfile:
```diff
@@ -138,46 +141,87 @@ pipeline {
-        stage('Build image') {
+        stage('Build images') {
+            
             when {
                 branch pattern: "release/[0-9].[0-9].[0-9]", comparator: "REGEXP"
                
             }
```
@wannabelll please check this comment

flannooz replied:
Yeah, good point)

LanchevychMaxym commented on Jenkinsfile:
```diff
@@ -340,12 +408,61 @@ pipeline {
+
+    success {
+        script {
+            sendDiscordNotification('SUCCESS', 'Deployment pipeline completed successfully.')
```
Will this be executed even if the last successfully executed stage is rollback?

LanchevychMaxym commented on Jenkinsfile:
```diff
@@ -340,12 +408,61 @@ pipeline {
+
     }
+
 post { 
```
If I'm not wrong we expected to have Notifications only when we deploy/rollback or do anything else with Stage deployment, that's why I'm not sure about keeping all these notifications here in 'post'
Please verify if this meets original ticket requirements

LanchevychMaxym commented on Streetcode/DbUpdate/Program.cs:
```diff
@@ -8,58 +9,66 @@ static int Main(string[] args)
-                .WithScriptsFromFileSystem(pathToSqlScripts)
-                .LogToConsole()
-                .Build();
+        Console.WriteLine($"Connection string: {connectionString}");
```
@ZhmudAnastasiia @IceStorman @ihor-yatsyshyn @yaroslav-vasylyshyn please help @wannabelll to mask password here, so we are safe to log this connection string.

flannooz replied:
maybe I have found a solution.

LanchevychMaxym commented on Streetcode/Streetcode.DAL/Persistence/ScriptsMigration/HealthCheck.sql:
We don't need to commit this test file. Let's remove it

flannooz replied:
ok

LanchevychMaxym commented on Streetcode/Streetcode.DAL/Persistence/ScriptsMigration/ZTest.sql:
Let's remove this as well

LanchevychMaxym commented on entrypoint.sh:
Is this file still needed?

flannooz replied:
this question needs research

flannooz committed:
Update Dockerfile.dbupdate

deleted "RUN ls -la /tmp/ScriptsMigration"

Co-authored-by: Maxym <47561209+LanchevychMaxym@users.noreply.github.com>

flannooz committed:
Update Jenkinsfile

Switched from the 'feature/add-init-container' branch to 'main'

flannooz committed:
Update Jenkinsfile

Fix REGEXP pattern for matching release branch names

flannooz committed:
Update Program.cs

removed "connection string"

flannooz committed:
Delete Streetcode/Streetcode.DAL/Persistence/ScriptsMigration/HealthCheck.sql

deleted "HealthCheck.sql" file

flannooz committed:
Delete Streetcode/Streetcode.DAL/Persistence/ScriptsMigration/ZTest.sql

deleted "ZTest.sql" file

flannooz committed:
Update Jenkinsfile

updated the Discord notifications logic to execute only when the status update about the deployment on the STG environment is available

flannooz committed:
Update Jenkinsfile at "Setup dependencies" stage

Resolved an issue where the pipeline failed at dotnet tool update due to dotnet-coverage version 17.13.1 already being installed
